testsuites/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ComplexSPARQLQueryTest.java

```diff
@@ -278,6 +279,36 @@ public void testNullContext2() throws Exception {
 		}
 	}
 
+	@Test
+	public void testSesameNilAsGraph() throws Exception {
+		loadTestData("/testdata-query/dataset-query.trig");
+		StringBuilder query = new StringBuilder();
+		query.append(getNamespaceDeclarations());
+		query.append(" SELECT * ");
+		query.append(" WHERE { GRAPH sesame:nil { ?s ?p ?o } } ");
+//		query.append(" WHERE { ?s ?p ?o } ");
+
+		TupleQuery tq = conn.prepareTupleQuery(QueryLanguage.SPARQL, query.toString());
+
+		try {
+			List<BindingSet> result = QueryResults.asList(tq.evaluate());
+
+			// nil graph should not be empty
+			assertThat(result.size()).isGreaterThan(1);
```

Comment:
could you just state the expected size instead?

Reply:
Actually no, because (potentially) the result will contain more statements on a store with inferencing enabled. We could of course disable that for the test, but this seemed simpler.